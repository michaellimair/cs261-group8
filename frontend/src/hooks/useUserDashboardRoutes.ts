import { dashboardRoutes, IDashboardRoute } from 'routes';
import useUser from 'hooks/useUser';

const useUserDashboardRoutes = (): IDashboardRoute[] => {
  const { user } = useUser();

  if (!user) {
    return [];
  }

  return dashboardRoutes
    .filter(({ allowedGroups }) => allowedGroups
      .some((groupName) => user.groups.find(({ name }) => name === groupName)));
};

export default useUserDashboardRoutes;
