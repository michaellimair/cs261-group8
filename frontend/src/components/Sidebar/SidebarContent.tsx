import { FC } from 'react';
import {
  Box,
  CloseButton,
  Flex,
  useColorModeValue,
  Text,
  BoxProps,
} from '@chakra-ui/react';
import SidebarNavItem from 'components/Sidebar/SidebarNavItem';
import useUserDashboardRoutes from 'hooks/useUserDashboardRoutes';
import { useTranslation } from 'react-i18next';

interface ISidebarContentProps extends BoxProps {
  onClose: () => void;
}

const SidebarContent: FC<ISidebarContentProps> = ({ onClose, ...rest }) => {
  const routes = useUserDashboardRoutes(true);
  const { t } = useTranslation();

  return (
    <Box
      transition="3s ease"
      bg={useColorModeValue('white', 'gray.900')}
      borderRight="1px"
      borderRightColor={useColorModeValue('gray.200', 'gray.700')}
      w={{ base: 'full', md: 60 }}
      pos="fixed"
      h="full"
      {...rest}
    >
      <Flex h="20" alignItems="center" mx="8" justifyContent="space-between">
        <Text fontSize="2xl" fontFamily="monospace" fontWeight="bold">
          Logo
        </Text>
        <CloseButton data-testid="closeButton" display={{ base: 'flex', md: 'none' }} onClick={onClose} />
      </Flex>
      {routes.map((route) => (
        <>
          <SidebarNavItem key={route.name} icon={route.icon} route={route}>
            {t(`dashboard.${route.name}.title`)}
          </SidebarNavItem>
          {route.children && (
          <Box ml="8">
            {route.children.map((child) => (
              <SidebarNavItem key={child.name} icon={child.icon} route={child}>
                {t(`dashboard.${child.name}.title`)}
              </SidebarNavItem>
            ))}
          </Box>
          )}
        </>
      ))}
    </Box>
  );
};

export default SidebarContent;
