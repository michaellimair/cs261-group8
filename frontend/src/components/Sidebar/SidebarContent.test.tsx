import { render } from '@testing-library/react';
import useUserDashboardRoutes from 'hooks/useUserDashboardRoutes';
import { BrowserRouter } from 'react-router-dom';
import { dashboardRoutes } from 'routes';
import SidebarContent from './SidebarContent';

jest.mock('hooks/useUserDashboardRoutes');

describe('SidebarContent', () => {
  let onClose: jest.Mock;

  beforeEach(() => {
    onClose = jest.fn();
    (useUserDashboardRoutes as jest.Mock).mockImplementationOnce(() => dashboardRoutes);
  });

  it('renders properly', () => {
    const result = render(<BrowserRouter><SidebarContent onClose={onClose} /></BrowserRouter>);
    expect(result).toMatchSnapshot();
  });

  it('invokes onClose function when the close button is clicked', () => {
    const result = render(<BrowserRouter><SidebarContent onClose={onClose} /></BrowserRouter>);
    const closeButton = result.queryByTestId('closeButton')!;

    expect(closeButton).not.toBeNull();

    closeButton.click();

    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
