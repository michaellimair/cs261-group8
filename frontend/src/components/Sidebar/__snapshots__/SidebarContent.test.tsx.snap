// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarContent renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-ltylk8"
      >
        <div
          class="css-1bxj9j3"
        >
          <p
            class="chakra-text css-1gfp23t"
          >
            Logo
          </p>
          <button
            aria-label="Close"
            class="css-33zxup"
            data-testid="closeButton"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <a
          class="chakra-link css-jgup0q"
          href="/welcome"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            dashboard.welcome.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentor-rec"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            dashboard.mentor-rec.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/dashboard"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
              <polyline
                points="9 22 9 12 15 12 15 22"
              />
            </svg>
            dashboard.home.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentor-only"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 22s8-4 8-10V5l-8-3-8 3v7c0 6 8 10 8 10z"
              />
            </svg>
            dashboard.mentor_only.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentee-only"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 368a16 16 0 01-7.94-2.11L108 285.84a8 8 0 00-12 6.94V368a16 16 0 008.23 14l144 80a16 16 0 0015.54 0l144-80a16 16 0 008.23-14v-75.22a8 8 0 00-12-6.94l-140.06 80.05A16 16 0 01256 368z"
              />
              <path
                d="M495.92 190.5v-.11a16 16 0 00-8-12.28l-224-128a16 16 0 00-15.88 0l-224 128a16 16 0 000 27.78l224 128a16 16 0 0015.88 0L461 221.28a2 2 0 013 1.74v144.53c0 8.61 6.62 16 15.23 16.43A16 16 0 00496 368V192a14.76 14.76 0 00-.08-1.5z"
              />
            </svg>
            dashboard.mentee_only.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentee-meetings"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
              />
            </svg>
            dashboard.mentee_meetings.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentor-meetings"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
              />
            </svg>
            dashboard.mentor_meetings.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/mentee-milestones"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 368a16 16 0 01-7.94-2.11L108 285.84a8 8 0 00-12 6.94V368a16 16 0 008.23 14l144 80a16 16 0 0015.54 0l144-80a16 16 0 008.23-14v-75.22a8 8 0 00-12-6.94l-140.06 80.05A16 16 0 01256 368z"
              />
              <path
                d="M495.92 190.5v-.11a16 16 0 00-8-12.28l-224-128a16 16 0 00-15.88 0l-224 128a16 16 0 000 27.78l224 128a16 16 0 0015.88 0L461 221.28a2 2 0 013 1.74v144.53c0 8.61 6.62 16 15.23 16.43A16 16 0 00496 368V192a14.76 14.76 0 00-.08-1.5z"
              />
            </svg>
            dashboard.mentee_milestones.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/feedbacks/create"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            dashboard.create_feedback.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/feedbacks/:id"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            dashboard.view_feedback.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/feedbacks/:id/edit"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            dashboard.edit_feedback.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/feedbacks"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.549 10.078c.46.182.88.424 1.258.725.378.3.701.65.97 1.046a4.829 4.829 0 0 1 .848 2.714V15H9.75v-.438a3.894 3.894 0 0 0-1.155-2.782 4.054 4.054 0 0 0-1.251-.84 3.898 3.898 0 0 0-1.532-.315A3.894 3.894 0 0 0 3.03 11.78a4.06 4.06 0 0 0-.84 1.251c-.206.474-.31.985-.315 1.531V15H1v-.438a4.724 4.724 0 0 1 .848-2.713 4.918 4.918 0 0 1 2.229-1.77 2.994 2.994 0 0 1-.555-.493 3.156 3.156 0 0 1-.417-.602 2.942 2.942 0 0 1-.26-.683 3.345 3.345 0 0 1-.095-.739c0-.423.08-.82.24-1.189a3.095 3.095 0 0 1 1.626-1.627 3.067 3.067 0 0 1 2.386-.007 3.095 3.095 0 0 1 1.627 1.627 3.067 3.067 0 0 1 .157 1.928c-.06.237-.148.465-.266.684a3.506 3.506 0 0 1-.417.608c-.16.187-.345.35-.554.492zM5.812 9.75c.301 0 .584-.057.848-.17a2.194 2.194 0 0 0 1.162-1.163c.119-.269.178-.554.178-.854a2.138 2.138 0 0 0-.643-1.538 2.383 2.383 0 0 0-.697-.472 2.048 2.048 0 0 0-.848-.178c-.3 0-.583.057-.847.17a2.218 2.218 0 0 0-1.17 1.17c-.113.264-.17.547-.17.848 0 .3.057.583.17.847.115.264.27.497.466.697a2.168 2.168 0 0 0 1.552.643zM15 1v7h-1.75l-2.625 2.625V8H9.75v-.875h1.75v1.388l1.388-1.388h1.237v-5.25h-8.75v1.572a7.255 7.255 0 0 0-.438.069 2.62 2.62 0 0 0-.437.123V1H15z"
              />
            </svg>
            dashboard.feedback.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/profile"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              />
            </svg>
            dashboard.profile.title
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/my-calendar"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              data-testid="sidebarNavIcon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
              />
            </svg>
            dashboard.my_calendar.title
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-ltylk8"
    >
      <div
        class="css-1bxj9j3"
      >
        <p
          class="chakra-text css-1gfp23t"
        >
          Logo
        </p>
        <button
          aria-label="Close"
          class="css-33zxup"
          data-testid="closeButton"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <a
        class="chakra-link css-jgup0q"
        href="/welcome"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          dashboard.welcome.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentor-rec"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          dashboard.mentor-rec.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/dashboard"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="none"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
            <polyline
              points="9 22 9 12 15 12 15 22"
            />
          </svg>
          dashboard.home.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentor-only"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="none"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 22s8-4 8-10V5l-8-3-8 3v7c0 6 8 10 8 10z"
            />
          </svg>
          dashboard.mentor_only.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentee-only"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 368a16 16 0 01-7.94-2.11L108 285.84a8 8 0 00-12 6.94V368a16 16 0 008.23 14l144 80a16 16 0 0015.54 0l144-80a16 16 0 008.23-14v-75.22a8 8 0 00-12-6.94l-140.06 80.05A16 16 0 01256 368z"
            />
            <path
              d="M495.92 190.5v-.11a16 16 0 00-8-12.28l-224-128a16 16 0 00-15.88 0l-224 128a16 16 0 000 27.78l224 128a16 16 0 0015.88 0L461 221.28a2 2 0 013 1.74v144.53c0 8.61 6.62 16 15.23 16.43A16 16 0 00496 368V192a14.76 14.76 0 00-.08-1.5z"
            />
          </svg>
          dashboard.mentee_only.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentee-meetings"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
            />
          </svg>
          dashboard.mentee_meetings.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentor-meetings"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
            />
          </svg>
          dashboard.mentor_meetings.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/mentee-milestones"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 368a16 16 0 01-7.94-2.11L108 285.84a8 8 0 00-12 6.94V368a16 16 0 008.23 14l144 80a16 16 0 0015.54 0l144-80a16 16 0 008.23-14v-75.22a8 8 0 00-12-6.94l-140.06 80.05A16 16 0 01256 368z"
            />
            <path
              d="M495.92 190.5v-.11a16 16 0 00-8-12.28l-224-128a16 16 0 00-15.88 0l-224 128a16 16 0 000 27.78l224 128a16 16 0 0015.88 0L461 221.28a2 2 0 013 1.74v144.53c0 8.61 6.62 16 15.23 16.43A16 16 0 00496 368V192a14.76 14.76 0 00-.08-1.5z"
            />
          </svg>
          dashboard.mentee_milestones.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/feedbacks/create"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          dashboard.create_feedback.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/feedbacks/:id"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          dashboard.view_feedback.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/feedbacks/:id/edit"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          dashboard.edit_feedback.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/feedbacks"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.549 10.078c.46.182.88.424 1.258.725.378.3.701.65.97 1.046a4.829 4.829 0 0 1 .848 2.714V15H9.75v-.438a3.894 3.894 0 0 0-1.155-2.782 4.054 4.054 0 0 0-1.251-.84 3.898 3.898 0 0 0-1.532-.315A3.894 3.894 0 0 0 3.03 11.78a4.06 4.06 0 0 0-.84 1.251c-.206.474-.31.985-.315 1.531V15H1v-.438a4.724 4.724 0 0 1 .848-2.713 4.918 4.918 0 0 1 2.229-1.77 2.994 2.994 0 0 1-.555-.493 3.156 3.156 0 0 1-.417-.602 2.942 2.942 0 0 1-.26-.683 3.345 3.345 0 0 1-.095-.739c0-.423.08-.82.24-1.189a3.095 3.095 0 0 1 1.626-1.627 3.067 3.067 0 0 1 2.386-.007 3.095 3.095 0 0 1 1.627 1.627 3.067 3.067 0 0 1 .157 1.928c-.06.237-.148.465-.266.684a3.506 3.506 0 0 1-.417.608c-.16.187-.345.35-.554.492zM5.812 9.75c.301 0 .584-.057.848-.17a2.194 2.194 0 0 0 1.162-1.163c.119-.269.178-.554.178-.854a2.138 2.138 0 0 0-.643-1.538 2.383 2.383 0 0 0-.697-.472 2.048 2.048 0 0 0-.848-.178c-.3 0-.583.057-.847.17a2.218 2.218 0 0 0-1.17 1.17c-.113.264-.17.547-.17.848 0 .3.057.583.17.847.115.264.27.497.466.697a2.168 2.168 0 0 0 1.552.643zM15 1v7h-1.75l-2.625 2.625V8H9.75v-.875h1.75v1.388l1.388-1.388h1.237v-5.25h-8.75v1.572a7.255 7.255 0 0 0-.438.069 2.62 2.62 0 0 0-.437.123V1H15z"
            />
          </svg>
          dashboard.feedback.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/profile"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
          dashboard.profile.title
        </div>
      </a>
      <a
        class="chakra-link css-jgup0q"
        href="/my-calendar"
        style="text-decoration: none;"
      >
        <div
          class="css-cpuzs9"
          role="group"
        >
          <svg
            class="chakra-icon css-1b1fm3x"
            data-testid="sidebarNavIcon"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
            />
          </svg>
          dashboard.my_calendar.title
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
